### PR TITLE
add: textmagic service

### PR DESCRIPTION
**Pull request by himsngh**
#82 
(Textmagic wasn't available in my country so I was not able to test it. Hope that's fine and if there are any issues please do let me know I will try to solve that)

### CODE REVIEW

File: service/textmagic/textmagic.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package textmagic
+
+import (
+	"context"
+	tm "github.com/textmagic/textmagic-rest-go-v2/v2"
```
Comment (nikoksr) [inline service/textmagic/textmagic.go:5]:
```suggestion
	textMagic "github.com/textmagic/textmagic-rest-go-v2/v2"
```

File: service/textmagic/textmagic.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"strings"
+)
+
+//TMagicService allow you to configure a TextMagic SDK client.
+type TMagicService struct {
+	UserName           string
+	APIKey             string
+	destinations       []string
+	TextMagicAPIClient *tm.APIClient
+}
```
Comment (nikoksr) [inline service/textmagic/textmagic.go:15]:
```suggestion
// Service allow you to configure a TextMagic SDK client.
type Service struct {
	userName           string
 	apiKey             string
	phoneNumbers       []string
 	client *textMagic.APIClient
}
```

File: service/textmagic/textmagic.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	config := tm.NewConfiguration()
+	client := tm.NewAPIClient(config)
+
+	return &TMagicService{
+		TextMagicAPIClient: client,
+		UserName:           userName,
+		APIKey:             apiKey,
+	}
+}
```
Comment (nikoksr) [inline service/textmagic/textmagic.go:28]:
```suggestion
// New creates a new text magic client. Use your user-name and API key from
// https://my.textmagic.com/online/api/rest-api/keys.
func New(userName, apiKey string) *Service {
	config := textMagic.NewConfiguration()
	client := textMagic.NewAPIClient(config)

	return &Service{
		client: client,
		userName:           userName,
		apiKey:             apiKey,
	}
}
```

File: service/textmagic/textmagic.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+// AddReceivers adds the given destination phone numbers to the notifier.
+func (s *TMagicService) AddReceivers(phoneNumbers ...string) {
+	s.destinations = append(s.destinations, phoneNumbers...)
+}
```
Comment (nikoksr) [inline service/textmagic/textmagic.go:33]:
```suggestion
// AddReceivers adds the given phone numbers to the notifier.
func (s *Service) AddReceivers(phoneNumbers ...string) {
	s.phoneNumbers = append(s.phoneNumbers, phoneNumbers...)
}
```

File: service/textmagic/textmagic.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	_, _, err := s.TextMagicAPIClient.TextMagicApi.SendMessage(auth, tm.SendMessageInputObject{
+		Text:   text,
+		Phones: strings.Join(s.destinations, ","),
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (nikoksr) [inline service/textmagic/textmagic.go:54]:
```suggestion
// Send sends a SMS via textmagic to all previously added receivers.
func (s *Service) Send(ctx context.Context, subject, message string) error {
	auth := context.WithValue(ctx, textMagic.ContextBasicAuth, textMagic.BasicAuth{
		UserName: s.userName,
		Password: s.apiKey,
	})

	text := subject + "\n" + message
	_, _, err := s.textMagicAPIClient.TextMagicApi.SendMessage(auth, textMagic.SendMessageInputObject{
		Text:   text,
		Phones: strings.Join(s.phoneNumbers, ","),
	})

	return err
}
```

File: service/textmagic/textmagic.go (hunk 1 above)
Comment (nikoksr) [reply service/textmagic/textmagic.go:5]: Make package name more expressive.

File: service/textmagic/textmagic.go (hunk 2 above)
Comment (nikoksr) [reply service/textmagic/textmagic.go:15]: Core struct should be called `Service` and none of its fields need to be exported.

File: service/textmagic/textmagic.go (hunk 3 above)
Comment (nikoksr) [reply service/textmagic/textmagic.go:28]: Function should just be called `New`; `textmagic.New()` is less redundant than `textmagic.NewTextMagicClient()`. Also, applying the new field names and moving comment about username and api key from `Send` method to here.

File: service/textmagic/textmagic.go (hunk 5 above)
Comment (nikoksr) [reply service/textmagic/textmagic.go:54]: We're already receiving a `context.Context` from the method's arguments and thus don't need to create a new one via `context.Background()`. 

Additionally apply new field names of `Service` struct and remove unnecessary error evaluation at the end of the method.

File: service/textmagic/Readme.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  notifier := notify.New()
+  notifier.UseServices(textMagic)
+
+  err := notifier.Send(context.Background(), "subject", "message")
+  if err != nil {
+    log.Fatalf("notifier.Send() failed: %s", err.Error())
+  }
+
+  log.Printf("notification sent")
```
Comment (nikoksr) [inline service/textmagic/Readme.md:35]:
```suggestion
  textMagicService := textmagic.New("YOUR_USER_NAME", "YOUR_API_KEY")

  textMagicService.AddReceivers("Destination1-Phone-Number")

  notifier := notify.New()
  notifier.UseServices(textMagicService)

  err := notifier.Send(context.Background(), "subject", "message")
  if err != nil {
    log.Fatalf("notifier.Send() failed: %v", err)
  }

  log.Printf("notification sent")
```

Comment (nikoksr) [reply service/textmagic/Readme.md:35]: Apply new function names and do a little clean-up.

**Review by nikoksr — CHANGES_REQUESTED** [review]
@himsngh I am very grateful to you for your contribution. It really helps us. There are unfortunately a few changes that need to be done and we definitely need to test the service before we can merge it. I'll see if I can find time to do that, otherwise it would be really great if someone else could do it.

**Comment by himsngh** [comment]
> @himsngh I am very grateful to you for your contribution. It really helps us. There are unfortunately a few changes that need to be done and we definitely need to test the service before we can merge it. I'll see if I can find time to do that, otherwise it would be really great if someone else could do it.

@nikoksr  I have done the changes.

**Review by nikoksr — APPROVED** [review]
Appreciate the changes and your contribution!

**Pull request merged**